### PR TITLE
fix: add support for no colors mode

### LINKED ISSUE #218: A "no formatting" mode for CI runs which don't support text formatting
**Is your feature request related to a problem? Please describe.**

Snapshot failures in my Jenkins runs show the colour codes which make it hard to read the diffs.

image

**Describe the solution you'd like**

An opt-in `--no-formatting` (`--no-colors`) mode.

**Describe alternatives you've considered**

Using a better CI runner.

**Additional context**

jest has a `--no-colors` mode.

### PR DESCRIPTION
**Pull request by iamogbz**
## Description

Adds support no colors mode with environment variables `NO_COLOR`, `ANSI_COLORS_DISABLED` or pytest option.

## Related Issues

- Closes #218 

## Checklist

- [x] This PR has sufficient documentation.
- [x] This PR has sufficient test coverage.
- [x] This PR title satisfies semantic [convention](https://www.conventionalcommits.org/en/v1.0.0/#summary).

## Additional Comments

- pytest supports by default `NO_COLOR`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,11 +74,12 @@ def __repr__(self) -> str:
...
+| ------------------------------ | ------------------------------------------------------------------------------------------------------------------------------ | ------------------------------------------------ |
+| `--snapshot-update`            | Snapshots will be updated to match assertions and unused snapshots will be deleted.                                            | `False`                                          |
+| `--snapshot-warn-unused`       | Prints a warning on unused snapshots rather than fail the test suite.                                                          | `False`                                          |
+| `--snapshot-default-extension` | Use to change the default snapshot extension class.                                                                            | `syrupy.extensions.amber.AmberSnapshotExtension` |
+| `--snapshot-no-colors`         | Disable test results output highlighting. Equivalent to setting the environment variables `ANSI_COLORS_DISABLED` or `NO_COLOR` | Disabled by default if not in terminal.          |
```
Comment (noahnu) [inline README.md:82]: Does pytest also natively support `ANSI_COLORS_DISABLED`?

Comment (iamogbz) [reply README.md:82]: > pytest supports by default `NO_COLOR`

So added both

File: src/syrupy/terminal.py [hunk 2]
```diff
@@ -1,26 +1,42 @@
...
+)
 
 import colored
 
+from .constants import DISABLE_COLOR_ENV_VARS
+from .utils import get_env_value
+
+
+def _is_color_disabled() -> bool:
+    return any(map(get_env_value, DISABLE_COLOR_ENV_VARS))
```
Comment (noahnu) [inline src/syrupy/terminal.py:13]: I'm fine with this (and actually prefer it) but I thought we had a lint rule around preferring comprehension over map/reduce etc

**Review by noahnu — APPROVED** [review]
LGTM

File: src/syrupy/terminal.py (hunk 2 above)
Comment (iamogbz) [reply src/syrupy/terminal.py:13]: I don't think there's a lint rule around it, just a general case by case basis imo

**Pull request merged**